opentelekomcloud/services/ces/resource_opentelekomcloud_ces_alarmrule.go

```diff
@@ -371,14 +350,16 @@ func resourceAlarmRuleRead(ctx context.Context, d *schema.ResourceData, meta int
 		return fmterr.Errorf(errCreationClient, err)
 	}
 
-	r, err := alarmrule.Get(client, d.Id()).Extract()
+	r, err := alarms.ShowAlarm(client, d.Id())
 	if err != nil {
 		return common.CheckDeletedDiag(d, err, "alarmrule")
 	}
-	log.Printf("[DEBUG] Retrieved alarm rule %s: %#v", d.Id(), r)
+	alarm := r[0]
```

Comment:
Slice length should be checked first

Reply:
Should it really be checked? The API always returns a list, otherwise it fails, here's the doc:
https://docs.otc.t-systems.com/api/ces/ces_03_0031.html
It's pretty bad and have only closing bracket but I think you get the point

Reply:
In this case it's working correctly, but checking slice before addressing it is a good practice 🙂 you can assume my comment a style recommendation, not a found problem.